### ISSUE TITLE
Add @available(iOS 16, *) to generated class

### ISSUE THREAD

**Issue by nathansalaun-sofia** [Kolos65/Mockable#67]
When using `@Mockable` on my project I have the following error: `Runtime support for parameterized protocol types is only available in iOS 16.0.0 or newer`. My app targets iOS 15.

It's not a problem in test targets because I only run tests on the latest iOS version. However in order to compile, the mock needs `@available(iOS 16, *)`.

**Comment by Kolos65** [comment]
Can you show the protocol you encountered this with?

**Comment by nathansalaun-sofia** [comment]
I can't show you the whole protocol for privacy reasons but I think it comes from the following properties:

```swift
  var foo: any SomeProtocol<SomeStruct> { get }
  var bar: any SomeProtocol<AnotherStruct> { get }
  
  var baz: any AnotherProtocol<SomeStruct> { get }
```

I guess the runtime support for `any` isn't full until iOS 16 for some cases, and the generated mock falls into those cases.

**Comment by nathansalaun-sofia** [comment]
The compiler asks me to add the `@available` attribute on the generated class though so surely it must work?

What's missing from iOS 15 is _runtime_ support, which doesn't have anything to do with the SDK used to compile the app. 

iOS 15 has partial runtime support for primary associated types and inside the app, we do not (because we can't) use patterns that require iOS 16 runtime support. But the rest of the app works fine, we just can't use primary associated types in function parameters (I believe), which is what the mock does.

The right thing to do would be to have the macro detect protocols with primary associated types and add the annotation on the class if there are any, since the generated code requires the iOS 16 _runtime_ in that case.

**Comment by Kolos65** [comment]
I see, you are right... Will try to write the code for detecting this edge case. Will request a review from you when its ready.

**Comment by Kolos65** [comment]
@nathansalaun-sofia Pls check out my PR if you can at: #68

**Comment by Kolos65** [comment]
@nathansalaun-sofia The fix is available in [0.0.10](https://github.com/Kolos65/Mockable/releases/tag/0.0.10)
